closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -643,11 +643,21 @@ find_crypto_library(jboolean traceEnabled, const char *chomepath)
 
     /* If JAVA_HOME is not null or empty and no library has been loaded yet, try there. */
     if ((NULL != chomepath) && ('\0' != *chomepath) && (NULL == crypto_library)) {
-#if defined(_WIN32)
-        static const char pathSuffix[] = "\\bin\\";
-#else /* defined(_WIN32) */
+#if defined(__APPLE__)
         static const char pathSuffix[] = "/lib/";
-#endif /* defined(_WIN32) */
+#elif defined(_WIN32) /* defined(__APPLE__) */
+        static const char pathSuffix[] = "\\bin\\";
+#elif defined(__aarch64__) /* defined(_WIN32) */
+        static const char pathSuffix[] = "/lib/aarch64/";
+#elif defined(__ppc64__) && defined(_AIX) /* defined(__aarch64__) */
+        static const char pathSuffix[] = "/lib/ppc64/";
+#elif defined(__PPC64__) && defined(__linux__) /* defined(__ppc64__) && defined(_AIX) */
+        static const char pathSuffix[] = "/lib/ppc64le/";
+#elif defined(__s390x__) /* defined(__PPC64__) && defined(__linux__) */
+        static const char pathSuffix[] = "/lib/s390x/";
+#else /* defined(__s390x__) */
+        static const char pathSuffix[] = "/lib/amd64/";
+#endif /* defined(__APPLE__) */
```

Comment:
This should use `OPENJDK_TARGET_CPU_LIBDIR` instead of duplicating the logic to determine the subdirectory.
```c
#if defined(_WIN32)
        static const char pathSuffix[] = "\\bin\\";
#else /* defined(_WIN32) */
        static const char pathSuffix[] = "/lib" OPENJDK_TARGET_CPU_LIBDIR "/";
#endif /* defined(_WIN32) */
```
This approach is enabled by adding to `CFLAGS` for `BUILD_JNCRYPTO` in `SecurityLibraries.gmk`:
```
            -DOPENJDK_TARGET_CPU_LIBDIR='"$(OPENJDK_TARGET_CPU_LIBDIR)"' \
```